cubit_shopper/lib/screens/catalog.dart

```diff
@@ -0,0 +1,120 @@
+// Copyright 2019 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import '../models/cart.dart';
+import '../models/catalog.dart';
+
+class MyCatalog extends StatelessWidget {
+  const MyCatalog({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: CustomScrollView(
+        slivers: [
+          _MyAppBar(),
+          const SliverToBoxAdapter(child: SizedBox(height: 12)),
+          SliverList(
+            delegate: SliverChildBuilderDelegate(
+                (context, index) => _MyListItem(index)),
+          ),
+        ],
+      ),
+    );
+  }
+}
+
+class _AddButton extends StatelessWidget {
+  final Item item;
+
+  const _AddButton({required this.item});
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocBuilder<CartModel, CartModelState>(
```

Comment:
Note this is less efficient than the Provider version, in that this version rebuilds all of the Add buttons, every time.